### PR TITLE
add kube-fluentd-operator

### PR DESCRIPTION
**Pull request by rawlingsj**
#### Quality Requirements for Images PRs
The items in this checklist should all be checked in the PR with exceptions clearly documented.
The general idea is that to the extent possible, the image should be a drop-in replacement to its public counterpart.
- [x] Image is a smaller size than its common public counterpart, or if not, the reason why should be explained
- [x] Image should be scanned for CVEs and should return 0 or near-0 results. Exceptions should be explained.
- [x] The software in the image should be able to be installed and work as expected
  - [x] The image should start in a cluster
  - [x] The application should be accessible to the user/cluster/etc. as expected
- [x] If there is an open source helm chart provided, the image should be able to be used with the chart. If needed, please add a -compat package to close any gaps with the public helm chart.
- [x] build for x86_64 and aarch64 if possible, document exceptions

#### For new image PRs only

If you have an apko.yaml file in this PR you need to follow this checklist, otherwise feel free to remove.
- [x] Include tests, sufficient enough that you would trust this image running in production.
- [x] Perform manual tests as needed and document your testing results
  - [ ] For services images (rabbit, databases, webservers) test basic functionality, upstream install/getting started, port availability, admin access. Document differences from public image.

- [x] The version included is the latest GA version of the software
- [x] The latest tag points to the newest stable version
- [ ] There is a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base')
- [ ] The image runs as `nonroot` and GID/UID are set to 65532 or upstream default
  - [ ] Alternatively the username and GID/UID may be a commonly used one from the ecosystem e.g: postgres
- [ ] The image contains the ca-certs bundle if needed
- [ ] ENTRYPOINT
  - [ ] For applications/servers/utilities call main program with no arguments e.g. [redis-server]
  - [ ] For base images leave empty
  - [ ] For dev variants set to entrypoint script that falls back to system
- [ ] CMD:
  - [x] For server applications give arguments to start in daemon mode (may be empty)
  - [ ] For utilities/tooling bring up help e.g. `–help`
  - [ ] For base images with a shell, call it e.g. [/bin/sh]

- [x] Add annotations e.g:
```
annotations:
  "org.opencontainers.image.url": https://edu.chainguard.dev/chainguard/chainguard-images/reference/busybox/ # use the academy site here
  "org.opencontainers.image.source": https://github.com/chainguard-images/images/tree/main/images/bazel # use github here
```
- [x] Check if environment variables are needed e.g. to set data locations
- [x] Ensure the image responds to SIGTERM
  - [x] `docker kill $(docker run -d --rm cgr.dev/chainguard/nginx)`
- [ ] Error logs write to stderr and normal logs to stdout. DO NOT write to file.
#### Documentation Requirements

- [x] Provide a README file, follow the [README template](readme-template.md) to include variants, tags, and usage examples
- [x] Document your testing and results in the PR
- [x] Document any exceptions to the testing and config requirements outlined in this template
- [x] Document where and how the image deviates from popular alternatives (ports, packages, user, entrypoint, versions, etc). Explain why there's a difference and if the user needs to do anything differently when using the image

### CODE REVIEW

**Comment by rawlingsj** [comment]
current CVE count
```
2 critical, 7 high, 12 medium, 1 low, 0 negligible
```

File: images/kube-fluentd-operator/tests/main.tf [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+data "oci_string" "ref" { input = var.digest }
+
+resource "random_pet" "suffix" {}
+
+resource "helm_release" "kfo" {
+  name             = "kfo-${random_pet.suffix.id}"
+  namespace        = "kfo"
+  chart            = "https://github.com/vmware/kube-fluentd-operator/releases/download/v1.17.4/log-router-0.4.0.tgz" # not sure how we can get the correct version of the chart rather than hardcoding a version below
```
Comment (rawlingsj) [inline images/kube-fluentd-operator/tests/main.tf:25]: this is the one thing I'm not sure about, it works for now but maintaining a version here isn't going to work really.  I couldn't figure out a combination of github `/latest` or `/release` that would work.

Comment (imjasonh) [reply images/kube-fluentd-operator/tests/main.tf:25]: Let's get this in and figure something out later when this breaks us.

File: images/kube-fluentd-operator/tests/main.tf [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  default     = false
+}
+
+variable "digest" {
+  description = "The image digest to run tests over."
+}
+
+data "oci_string" "ref" { input = var.digest }
+
+resource "random_pet" "suffix" {}
```
Comment (imjasonh) [inline images/kube-fluentd-operator/tests/main.tf:20]: You probably don't need a random name if there's only going to be one of these. It's a small thing, I won't block the PR on it.

File: images/kube-fluentd-operator/tests/main.tf [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    random = { source = "hashicorp/random" }
+  }
+}
+
+variable "skip_crds" {
+  description = "Used to deconflict between multiple installations within the same cluster."
+  default     = false
+}
+
+variable "digest" {
```
Comment (imjasonh) [inline images/kube-fluentd-operator/tests/main.tf:14]:
```suggestion
}

variable "digest" {
```

File: images/kube-fluentd-operator/tests/smoke-test.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+set -o errexit -o nounset -o pipefail -x
+
+sleep 5
+
+# There are docs to test a deployment of the operator, but this is not
+# working, same result with the upstream image. # https://github.com/vmware/kube-fluentd-operator#try-it-out
+
+# Instead, we will test the operator by deploying the log-router helm chart
+# and check the deamonset pod is healthy + the logs contain the expected text
```
Comment (imjasonh) [inline images/kube-fluentd-operator/tests/smoke-test.sh:11]: The helm_release will wait for its resources to become healthy already, FWIW. I think it's still useful to check the logs though.

**Comment by rawlingsj** [comment]
OK think this is ready now @imjasonh

**Review by imjasonh — APPROVED** [review]

**Pull request merged**